presto-orc/src/main/java/com/facebook/presto/orc/ByteArrayUtils.java

```diff
@@ -48,35 +48,44 @@ private ByteArrayUtils() {}
     // Adapted from io.airlift.slice.Slice
     public static int compareRanges(byte[] left, int leftOffset, int leftLength, byte[] right, int rightOffset, int rightLength)
     {
-        int minLength = min(leftLength, rightLength);
-        int i = 0;
-        if (minLength >= SIZE_OF_LONG) {
-            while (i < minLength) {
-                long leftLong = unsafe.getLong(left, (long) ARRAY_BYTE_BASE_OFFSET + leftOffset + i);
-                long rightLong = unsafe.getLong(right, (long) ARRAY_BYTE_BASE_OFFSET + rightOffset + i);
+        long leftAddress = ARRAY_BYTE_BASE_OFFSET + leftOffset;
+        long rightAddress = ARRAY_BYTE_BASE_OFFSET + rightOffset;
 
-                if (leftLong != rightLong) {
-                    return longBytesToLong(leftLong) < longBytesToLong(rightLong) ? -1 : 1;
-                }
+        int compareLength = min(leftLength, rightLength);
+        while (compareLength >= SIZE_OF_LONG) {
+            long leftLong = unsafe.getLong(left, leftAddress);
+            long rightLong = unsafe.getLong(right, rightAddress);
 
-                i += SIZE_OF_LONG;
+            if (leftLong != rightLong) {
+                return longBytesToLong(leftLong) < longBytesToLong(rightLong) ? -1 : 1;
             }
-            i -= SIZE_OF_LONG;
+
+            leftAddress += SIZE_OF_LONG;
+            rightAddress += SIZE_OF_LONG;
+            compareLength -= SIZE_OF_LONG;
         }
 
-        while (i < minLength) {
-            int leftByte = Byte.toUnsignedInt(left[leftOffset + i]);
-            int rightByte = Byte.toUnsignedInt(right[rightOffset + i]);
+        while (compareLength > 0) {
+            byte leftByte = unsafe.getByte(left, leftAddress);
+            byte rightByte = unsafe.getByte(right, rightAddress);
 
-            if (leftByte != rightByte) {
-                return leftByte - rightByte;
+            int compareResult = compareUnsignedBytes(leftByte, rightByte);
```

Comment:
leftByte and rightByte is only used once in this line. Maybe just inline them here?